controllers/idler/idler_controller_test.go

```diff
@@ -682,51 +682,116 @@ func TestAppNameTypeForControllers(t *testing.T) {
-		assert.NoError(t, err)
-		assert.True(t, res.Requeue)
+	for pt, tcs := range testpod {
+		t.Run(pt, func(t *testing.T) {
+			reconciler, req, cl, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet, mur)
+			idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
 
-		// second reconcile should delete pods and create notification
-		res, err = reconciler.Reconcile(context.TODO(), req)
-		//then
-		assert.NoError(t, err)
-		memberoperatortest.AssertThatIdler(t, idler.Name, cl).
-			HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
-		//check the notification is actually created
-		hostCl, _ := reconciler.GetHostCluster()
-		notification := &toolchainv1alpha1.Notification{}
-		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
-			Namespace: test.HostOperatorNs,
-			Name:      "alex-stage-idled",
-		}, notification)
-		require.NoError(t, err)
-		require.Equal(t, "alex@test.com", notification.Spec.Recipient)
-		require.Equal(t, "idled", notification.Labels[toolchainv1alpha1.NotificationTypeLabelKey])
-		require.Equal(t, "Pod", notification.Spec.Context["AppType"])
-		require.Equal(t, p.Name, notification.Spec.Context["AppName"])
+			if tcs.controllerOwned {
+				preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond...)
+			} else {
+				p := preparePayloadsSinglePod(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond...).standalonePods[0]
+				pname = p.Name
+			}
 
-	})
+			// first reconcile to track pods
+			res, err := reconciler.Reconcile(context.TODO(), req)
+			assert.NoError(t, err)
+			assert.True(t, res.Requeue)
+
+			// second reconcile should delete pods and create notification
+			res, err = reconciler.Reconcile(context.TODO(), req)
+			//then
+			assert.NoError(t, err)
+
+			if tcs.expectedNotificationCreated {
+				memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+					HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
+			} else {
+				memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+					HasConditions(memberoperatortest.Running())
+			}
+			//check the notification is actually created
+			hostCl, _ := reconciler.GetHostCluster()
+			notification := &toolchainv1alpha1.Notification{}
+			err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
+				Namespace: test.HostOperatorNs,
+				Name:      "feny-stage-idled",
+			}, notification)
+			if tcs.expectedNotificationCreated {
+				require.NoError(t, err)
+				require.Equal(t, "feny@test.com", notification.Spec.Recipient)
+				require.Equal(t, "idled", notification.Labels[toolchainv1alpha1.NotificationTypeLabelKey])
+				require.Equal(t, tcs.expectedAppType, notification.Spec.Context["AppType"])
+				if !tcs.controllerOwned {
+					require.Equal(t, pname, notification.Spec.Context["AppName"])
```

Comment:
It would be nice to check the AppName for the controllers but I understand it's a bit tricky in this specific test. That's probably fine because AppNames is covered in the other tests for the controllers anyway.

Reply:
The Same thoughts and hence didn't add the tests for controller here 
Thanks for the review